### PR TITLE
Go: Refine Initialize Metrics

### PR DESCRIPTION
**Pull request by pellared**
Changes based on https://github.com/open-telemetry/opentelemetry-go/pull/4504 and https://github.com/open-telemetry/opentelemetry-go/pull/4540

### CODE REVIEW

File: content/en/docs/instrumentation/go/manual.md [hunk 1]
```diff
@@ -374,31 +374,39 @@ import (
...
-
 	// Handle shutdown properly so nothing leaks.
 	defer func() {
 		if err := meterProvider.Shutdown(context.Background()); err != nil {
 			log.Println(err)
 		}
 	}()
+
+	// Register as global meter provider so that it can
+	// that can used via otel.Meter and accessed using otel.GetMeterProvider.
```
Comment (dmathieu) [inline content/en/docs/instrumentation/go/manual.md:399]:
```suggestion
	// then be used via otel.Meter and accessed using otel.GetMeterProvider.
```

File: content/en/docs/instrumentation/go/manual.md [hunk 2]
```diff
@@ -374,31 +374,39 @@ import (
...
+	// Most instrumentation libraries use the global meter provider as default.
+	// If the global meter provider is not set then a no-op implementation
+	// is used and which fails to generate data.
```
Comment (dmathieu) [inline content/en/docs/instrumentation/go/manual.md:402]:
```suggestion
	// is used, which fails to generate data.
```

File: content/en/docs/instrumentation/go/manual.md (hunk 1 above)
Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md]: https://github.com/open-telemetry/opentelemetry.io/pull/3305/commits/45d6031ac453425eca05c9bc52d19f4d28086d5e

File: content/en/docs/instrumentation/go/manual.md (hunk 2 above)
Comment (pellared) [reply content/en/docs/instrumentation/go/manual.md]: https://github.com/open-telemetry/opentelemetry.io/pull/3305/commits/45d6031ac453425eca05c9bc52d19f4d28086d5e

File: content/en/docs/instrumentation/go/manual.md [hunk 3]
```diff
@@ -374,31 +374,39 @@ import (
 )
 
 func main() {
+  // Create resource.
 	res, err := newResource()
 	if err != nil {
 		panic(err)
```
Comment (cartermp) [inline content/en/docs/instrumentation/go/manual.md:380]:
```suggestion
  	// Create resource.
	res, err := newResource()
	if err != nil {
		panic(err)
```

**Review by cartermp — APPROVED** [review]
overall looks good, just a minor formatting thing

**Review by MrAlias — APPROVED** [review]

**Pull request merged**